deku-derive/src/macros/deku_read.rs

```diff
@@ -102,9 +106,16 @@ fn emit_enum(input: &DekuReceiver) -> Result<TokenStream, darling::Error> {
     let ident = &input.ident;
     let ident = quote! { #ident #ty };
 
+    // TODO: replace `expect` with an error.
     let id_type = input.id_type.as_ref().expect("expected `id_type` on enum");
-    let id_is_le_bytes = input.endian == EndianNess::Little;
-    let id_bit_size = super::option_as_literal_token(input.id_bits);
+
+    let id_is_le_bytes = input.endian.unwrap_or_default() == EndianNess::Little;
```

Comment:
To stay consistent with others:
```rust
    let id_is_le = input
        .endian
        .or(input.endian)
        .map(|endian| endian == EndianNess::Little)
```

Reply:
`unwrap_or_default()` and `or()` is different. the first one means "It could be None, and we will give it a default value". The second one means "We try it first, if None use other". And 
```rust
let id_is_le_bytes: bool = input.endian.unwrap_or_default() == EndianNess::Little;
let id_is_le: Option<bool> = input
        .endian
        .or(input.endian)
        .map(|endian| endian == EndianNess::Little)
```